code/game/data_huds.dm

```diff
@@ -281,32 +281,48 @@
 
 /mob/living/carbon/human/proc/sec_hud_set_security_status()
 	var/image/holder = hud_list[WANTED_HUD]
-	var/icon/I = icon(icon, icon_state, dir)
-	holder.pixel_y = I.Height() - world.icon_size
-	var/perpname = get_face_name(get_id_name(""))
-	if(perpname && GLOB.data_core)
-		var/datum/data/record/R = find_record("name", perpname, GLOB.data_core.security)
-		if(R)
-			switch(R.fields["criminal"])
-				if("Arrest")
-					holder.icon_state = "hudwanted"
-					return
-				if("Incarcerated")
-					holder.icon_state = "hudincarcerated"
-					return
-				if("Paroled")
-					holder.icon_state = "hudparolled"
-					return
-				if("Discharged")
-					holder.icon_state = "huddischarged"
-					return
-				if("Search")
-					holder.icon_state = "hudsearch"
-					return
-				if("Monitor")
-					holder.icon_state = "hudmonitor"
-					return
-	holder.icon_state = null
+	var/icon/sec_icon = icon(icon, icon_state, dir)
+	holder.pixel_y = sec_icon.Height() - world.icon_size
+	var/perp_name = get_face_name(get_id_name(""))
+
+	if(!perp_name || !GLOB.manifest)
+		holder.icon_state = null
+		return
+
+	var/datum/record/crew/target = find_record(perp_name)
+	if(!target)
+		return
+
+	switch(target.wanted_status)
+		if(WANTED_ARREST)
+			holder.icon_state = "hudwanted"
+		if(WANTED_PRISONER)
+			holder.icon_state = "hudincarcerated"
+		if(WANTED_SUSPECT)
+			holder.icon_state = "hudsuspected"
+		if(WANTED_PAROLE)
+			holder.icon_state = "hudparolled"
+		if(WANTED_DISCHARGED)
+			holder.icon_state = "huddischarged"
+		if(WANTED_NONE)
+			holder.icon_state = "hudnone"
+
+//Utility functions
+
+/**
+ * Updates the visual security huds on all mobs in GLOB.human_list that match the name passed to it.
+ */
+/proc/update_matching_security_huds(perp_name)
+	for (var/mob/living/carbon/human/h as anything in GLOB.human_list)
+		if (h.get_face_name(h.get_id_name("")) == perp_name)
```

Comment:
```suggestion
		if (h.get_face_name(h.get_id_name()) == perp_name)
```
`""` is kinda considered to be null/FALSE, that is overrided by the proc default parameter("unknown") which means `""` does nothing here.